doc/source/ray-client.rst

```diff
@@ -0,0 +1,65 @@
+**********
+Ray Client
+**********
+
+.. note::
+
+   This feature is still in beta and subject to changes.
+
+=====
+Basic usage
+=====
+
+While in beta, the server is available as an executable module. To start the server, run
+
+``python -m ray.util.client.server [--host host_ip] [--port port]``
+
+This runs ``ray.init()`` with default options and exposes the client gRPC port at ``host_ip:port`` (by default, ``0.0.0.0:50051``)
+
+From here, another Ray script can access that server from a networked machine with ``ray.connect()``
+
+.. code-block:: python
+   import ray
+   ray.connect("0.0.0.0:50051")  # replace with the appropriate host and port
+
+   # Normal Ray code follows
+   @ray.remote
+   def f(x):
+       return x ** x
+
+   do_work.remote(2)
+   #....
+  
+When the client disconnects, any object or actor references held by the server on behalf of the client are dropped, as if directly disconnecting from the cluster 
+
+
+====
+``RAY_CLIENT_MODE``
+====
+
+Because Ray client mode affects the behavior of the Ray API, larger scripts or libraries imported before ``ray.connect()`` may not realize they're in client mode.
```

Comment:
This is an issue we should fix right? How about adding a link to a github issue that deferred imports will be fixed in a later release?